lib/elixir/lib/calendar/date.ex

```diff
@@ -820,6 +821,37 @@ defmodule Date do
     %Date{year: year, month: month, day: day, calendar: target_calendar}
   end
 
+  defp new_duration!(%Duration{hour: 0, minute: 0, second: 0, microsecond: {0, 0}} = duration) do
+    duration
+  end
+
+  defp new_duration!(%Duration{}) do
+    raise ArgumentError, "duration may not contain time scale units"
+  end
```

Comment:
Since we already check for this on Calendar.ISO, should this simply be:

```suggestion
  defp new_duration!(%Duration{} = duration) do
    duration
  end
```

?

Same for time?

Reply:
Yes! For a moment i thought we could move the match one level up into the calendar types, but it's better to keep the guard rail in the calendar of course 👍